File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int len = arr.length;
        int second = 0;
        int largest = 0;
        for(int i=0;i<len ;i++){
            if(arr[i]>largest){
                second = largest;
                largest = arr[i];
            }
            else if(second<arr[i] && arr[i]!=largest){
                second=arr[i];
            }
        }
        if(second==0) return -1;
        
        return second;
    }
}

File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int length = arr.size();
        int max = 0;
        int min = 0;
        for(int i=0;i<length;i++){
            if(arr[i]>max){
                min = max;
                max = arr[i];
            }
            else if(arr[i]>min && arr[i]!=max){
                min = arr[i];
            }
        }
        if(min==0) return -1;
        
        return min;
        
    }
};